mpas_analysis/ocean/streamfunction_moc.py

```diff
@@ -338,6 +337,327 @@ def _load_mesh(self):  # {{{
+                    latBins = latCell[indRegion]
+                    latBins = np.arange(np.amin(latBins),
+                                        np.amax(latBins)+latBinSize,
+                                        latBinSize)
+                mocTop = self._compute_moc(latBins, nVertLevels, latCell,
+                                           regionCellMask, transportZ, velArea)
+            # hard-wire region=0 (Atlantic) for now
+            indRegion = 0
+            mocTop = dsLocal.timeMonthly_avg_mocStreamvalLatAndDepthRegion[timeIndex, indRegion, :, :].values
+            mocRegion[timeIndex] = np.amax(mocTop[:, indlat26])
+
+                # Store computed MOC to dictionary
+                self.lat[region] = latBins
+                self.moc[region] = mocTop
+# Needs to be edited till this point
+#*********************
+
+            # Save to file
+            self.logger.info('   Save global and regional MOC to file...')
+            ncFile = netCDF4.Dataset(outputFileClimo, mode='w')
+            # create dimensions
+            ncFile.createDimension('nz', len(refTopDepth))
+            for region in self.regionNames:
+                latBins = self.lat[region]
+                mocTop = self.moc[region]
+                ncFile.createDimension('nx{}'.format(region), len(latBins))
+                # create variables
+                x = ncFile.createVariable('lat{}'.format(region), 'f4',
+                                          ('nx{}'.format(region),))
+                x.description = 'latitude bins for MOC {}'\
+                                ' streamfunction'.format(region)
+                x.units = 'degrees (-90 to 90)'
+                y = ncFile.createVariable('moc{}'.format(region), 'f4',
+                                          ('nz', 'nx{}'.format(region)))
+                y.description = 'MOC {} streamfunction, annual'\
+                                ' climatology'.format(region)
+                y.units = 'Sv (10^6 m^3/s)'
+                # save variables
+                x[:] = latBins
+                y[:, :] = mocTop
+            depth = ncFile.createVariable('depth', 'f4', ('nz',))
+            depth.description = 'depth'
+            depth.units = 'meters'
+            depth[:] = refTopDepth
+            ncFile.close()
+        else:
+            # Read from file
+            self.logger.info('   Read previously computed MOC streamfunction '
+                             'from file...')
+            ncFile = netCDF4.Dataset(outputFileClimo, mode='r')
+            self.depth = ncFile.variables['depth'][:]
+            self.lat = {}
+            self.moc = {}
+            for region in self.regionNames:
+                self.lat[region] = ncFile.variables['lat{}'.format(region)][:]
+                self.moc[region] = \
+                    ncFile.variables['moc{}'.format(region)][:, :]
+            ncFile.close()
+        # }}}
+
+    def _compute_moc_time_series_analysismember(self):  # {{{
+
```

Comment:
This is mostly done, but we probably do not need to save the timeseries to file this time, since it's a super simple calculation once we have the online MOC.
Also, there are lots of repetitions with the `_compute_time_series_postprocess` function.

Reply:
Please do make sure to save the time series data.  The main reason is that it makes it possible to use it as a reference data set when doing a main vs. ref comparison.  In the not-too-distant future, we would want to do the same with the MOC lat vs. depth plots, too, so we will also need to store a copy of them.